src/main/java/edu/cornell/kfs/tax/service/impl/TaxProcessingV2ServiceImpl.java

```diff
@@ -0,0 +1,151 @@
+        final TaxBatchConfig taxConfig;
+        if (taxRangeSettings.size() == 1) {
+            final String taxRangeSetting = taxRangeSettings.iterator().next();
+            taxConfig = buildTaxBatchConfigForSingleRangeSetting(processingStartDate, taxType, taxRangeSetting);
+        } else if (taxRangeSettings.size() == 2) {
+            taxConfig = buildTaxBatchConfigForExplicitStartAndEndDates(processingStartDate, taxType, taxRangeSettings);
+        } else if (taxRangeSettings.size() == 0) {
+            throw new IllegalStateException("The dates-to-process parameter cannot be blank");
+        } else {
+            throw new IllegalStateException("The dates-to-process parameter should only have either 1 value or 2 "
+                    + "semicolon-delimited values");
+        }
+
+        LOG.info("buildTaxBatchConfig, Will perform {} processing for transactions in year {} between {} and {}",
+                taxConfig.getTaxType(), taxConfig.getReportYear(), taxConfig.getStartDate(), taxConfig.getEndDate());
+
+        return taxConfig;
+    }
+
+    private TaxBatchConfig buildTaxBatchConfigForSingleRangeSetting(final java.util.Date processingStartDate,
+            final String taxType, final String taxRangeSetting) {
+        final int reportYear = getReportYear(taxRangeSetting);
+        final LocalDate startDate = LocalDate.of(reportYear, 1, 1);
+        final LocalDate endDate = LocalDate.of(reportYear, 12, 31);
+        return new TaxBatchConfig(taxType, reportYear, processingStartDate,
+                java.sql.Date.valueOf(startDate), java.sql.Date.valueOf(endDate));
+    }
+
+    private int getReportYear(final String taxRangeSetting) {
+        if (StringUtils.equals(taxRangeSetting, CUTaxConstants.YEAR_TO_DATE)) {
+            return dateTimeService.getLocalDateNow().getYear();
+        } else if (StringUtils.equals(taxRangeSetting, CUTaxConstants.PREVIOUS_YEAR_TO_DATE)) {
+            return dateTimeService.getLocalDateNow().getYear() - 1;
+        } else {
+            return parseExplicitReportYear(taxRangeSetting);
+        }
+    }
+
+    private int parseExplicitReportYear(final String reportYear) {
+        try {
+            final int parsedYear = Integer.parseInt(reportYear);
+            if (parsedYear <= LocalDate.EPOCH.getYear() || parsedYear > 9999) {
+                throw new IllegalStateException("The dates-to-process parameter should have contained a 4-digit "
+                        + "tax year later than 1970");
+            }
+            return parsedYear;
+        } catch (final NumberFormatException e) {
+            throw new IllegalStateException("The dates-to-process parameter contained a malformed tax year");
+        }
+    }
+
+    private TaxBatchConfig buildTaxBatchConfigForExplicitStartAndEndDates(final java.util.Date processingStartDate,
+            final String taxType, final Collection<String> taxRangeSettings) {
```

Comment:
Will the taxType always be 1042-S? Does it make sense to build this process to allow for multiple tax types?

Reply:
At some point, we may want to refactor the 1099 code for simplicity and easier maintainability. If we do eventually proceed with such refactoring, it might make sense to add its basic setup portions into this service instead of creating a separate one.